lib/Sema/TypeCheckMacros.cpp

```diff
@@ -688,7 +689,15 @@ static void validateMacroExpansion(SourceFile *expansionBuffer,
             introducedNameSet.count(MacroDecl::getArbitraryName()));
   };
 
-  for (auto *decl : expansionBuffer->getTopLevelDecls()) {
+  for (auto item : expansionBuffer->getTopLevelItems()) {
+    auto *decl = item.dyn_cast<Decl *>();
+    if (!decl) {
+      auto &ctx = dc->getASTContext();
```

Comment:
Is the only point of `dc` that we use it to get the `ASTContext`? That's available via `expansionBuffer` and other parameters.